pkg/inject/report.go

```diff
@@ -27,21 +27,25 @@ func newReport(conf *ResourceConfig) Report {
 	var name string
 	if m := conf.pod.Meta; m != nil {
 		name = m.Name
+		if name == "" {
+			name = m.GenerateName
```

Comment:
Not sure how useful this will be, as it only returns the prefix of the pod name like `linkerd-controller-7c6b8c586b-`. Will there be cases where the pod name will be empty? The parent workload name, if non-empty, might be more useful.

Reply:
The pod name appears to be always empty in the admission request metadata, and we only have access to that `GenerateName`, which is actually the ReplicaSet's name (plus a dash). I think this is a bit more informative (`linkerd-controller-7c6b8c586b-`) than the Deployment's name (`linkerd-controller`).